### DISCUSSION TITLE
How to run agent as daemon?

### DISCUSSION THREAD

**Discussion by den-pro-soft** [PrefectHQ/prefect#3587] [General]
# Description
I made a flow perfectly and I'm gonna register it to cloud and run agent as daemon. My current code is like so:
```
        ...
        ...
        with Flow('S3TOSNOW Flow',
                  environment=LocalEnvironment(executor=DaskExecutor())
                  ) as flow:
            parsed_data = STS.read_data_s3.map(self.data_list)
            STS.load_data_to_snowflake.map(parsed_data)

        flow.register(project_name=self.project_name)
        flow.run_agent()
```

But this is just running agent until I quite the terminal. how can I leave it work?

**Comment by znicholasbrown** [comment]
Hi @aj-super - agents can be started from the CLI using `prefect agent start`, and can be passed the same configuration options you would when starting from Python; you can read more about [starting agents in the docs](https://docs.prefect.io/orchestration/agents/overview.html#overview). You can start an agent in the background using the method @madkinsz mentioned in [your thread from yesterday](https://github.com/PrefectHQ/prefect/discussions/3583), did that solution work for you? If so it would be good to respond there as well for others.

**Reply by den-pro-soft** [reply]
Then, what if I created 10 flows, do I have to run 10 agents in background? what if I created 20 flows? or we can run multiple flows on a single agent?
I would like to know the best way to keep running some registered flows on schedule? To get the flows work, you know, we have to keep running agents(or an agent) forever. How can I do this?

**Reply by zanieb** [reply]
An agent does not need to be created for each flow, they will pick up and run flows as they see them. You should only need a single agent to start. You would leave it running roughly "forever" and it will pick up flows as they are scheduled by the backend (server or cloud).

I think you would benefit from reading through the orchestration docs at https://docs.prefect.io/orchestration/

**Reply by den-pro-soft** [reply]
ah.. okay. Then, can you please let me know how to run the agent forever?
Is there any option that keeps it running as daemon in `run_agent`?

**Reply by zanieb** [reply]
You should run it from the CLI as Nicholas described, you can then use a `screen` as described at https://github.com/PrefectHQ/prefect/discussions/3583 or you could use something like `nohup`
